packages/graphql/src/translate/translate-top-level-match.ts

```diff
@@ -107,10 +103,17 @@ export function createMatchClause({
 
         preComputedWhereFieldSubqueries = preComputedSubqueries;
 
-        whereClause =
-            preComputedWhereFieldSubqueries && !preComputedWhereFieldSubqueries.empty
-                ? new Cypher.With("*")
-                : matchClause;
+        if (preComputedWhereFieldSubqueries && !preComputedWhereFieldSubqueries.empty) {
+            if (
+                preComputedWhereFieldSubqueries.children.length === 1 &&
+                preComputedWhereFieldSubqueries.children[0] instanceof Cypher.Match
+            ) {
+                whereClause = preComputedWhereFieldSubqueries.children[0];
+                preComputedWhereFieldSubqueries = undefined;
+            } else {
+                whereClause = new Cypher.With("*");
+            }
+        }
```

Comment:
This logic exists because I'm taking advantage of the cypher builder tree to decide wether or not to add a `WITH *`, which is not ideal

The nice solution would be fixing the problem we discussed regarding passing subqueries and queries without any context